/bin/bash /home/jinyao/PycharmProjects/IADBE/train_test_kolektor_draem.sh
(IADBE) jinyao@jinyao-System-Product-Name:~/PycharmProjects/IADBE$ /bin/bash /home/jinyao/PycharmProjects/IADBE/train_test_kolektor_draem.sh
Running command: anomalib train --data ./configs/data/kolektor.yaml --config ./configs/models/draem.yaml
2024-07-16 10:18:52,016 - anomalib.utils.config - WARNING - Anomalib currently does not support multi-gpu training. Setting devices to 1.
[07/16/24 10:18:52] WARNING  Anomalib currently does not support multi-gpu training. Setting devices to 1.                                                                                                    config.py:262
2024-07-16 10:18:52,032 - anomalib.models.components.base.anomaly_module - INFO - Initializing Draem model.
                    INFO     Initializing Draem model.                                                                                                                                                 anomaly_module.py:42
2024-07-16 10:18:52,541 - anomalib.callbacks - INFO - Loading the callbacks
                    INFO     Loading the callbacks                                                                                                                                                           __init__.py:43
2024-07-16 10:18:52,542 - anomalib.engine.engine - INFO - Overriding gradient_clip_val from None with 0 for Draem
                    INFO     Overriding gradient_clip_val from None with 0 for Draem                                                                                                                           engine.py:84
2024-07-16 10:18:52,543 - anomalib.engine.engine - INFO - Overriding num_sanity_val_steps from None with 0 for Draem
                    INFO     Overriding num_sanity_val_steps from None with 0 for Draem                                                                                                                        engine.py:84
2024-07-16 10:18:52,633 - lightning.pytorch.utilities.rank_zero - INFO - Trainer already configured with model summary callbacks: [<class 'lightning.pytorch.callbacks.rich_model_summary.RichModelSummary'>]. Skipping setting a default `ModelSummary` callback.
[07/16/24 10:18:52] INFO     Trainer already configured with model summary callbacks: [<class 'lightning.pytorch.callbacks.rich_model_summary.RichModelSummary'>]. Skipping setting a default               rank_zero.py:63
                             `ModelSummary` callback.
2024-07-16 10:18:52,641 - lightning.pytorch.utilities.rank_zero - INFO - GPU available: True (cuda), used: True
                    INFO     GPU available: True (cuda), used: True                                                                                                                                         rank_zero.py:63
2024-07-16 10:18:52,642 - lightning.pytorch.utilities.rank_zero - INFO - TPU available: False, using: 0 TPU cores
                    INFO     TPU available: False, using: 0 TPU cores                                                                                                                                       rank_zero.py:63
2024-07-16 10:18:52,643 - lightning.pytorch.utilities.rank_zero - INFO - IPU available: False, using: 0 IPUs
                    INFO     IPU available: False, using: 0 IPUs                                                                                                                                            rank_zero.py:63
2024-07-16 10:18:52,643 - lightning.pytorch.utilities.rank_zero - INFO - HPU available: False, using: 0 HPUs
                    INFO     HPU available: False, using: 0 HPUs                                                                                                                                            rank_zero.py:63
2024-07-16 10:18:52,644 - anomalib.models.components.base.anomaly_module - WARNING - No implementation of `configure_transforms` was provided in the Lightning model. Using default transforms from the base class. This may not be suitable for your use case. Please override `configure_transforms` in your model.
                    WARNING  No implementation of `configure_transforms` was provided in the Lightning model. Using default transforms from the base class. This may not be suitable for your use     anomaly_module.py:235
                             case. Please override `configure_transforms` in your model.
2024-07-16 10:18:52,646 - anomalib.data.image.kolektor - INFO - Found the dataset.
                    INFO     Found the dataset.                                                                                                                                                             kolektor.py:340
2024-07-16 10:18:52,689 - lightning.pytorch.utilities.rank_zero - INFO - You are using a CUDA device ('NVIDIA GeForce RTX 3090') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
                    INFO     You are using a CUDA device ('NVIDIA GeForce RTX 3090') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' |         rank_zero.py:63
                             'high')` which will trade-off precision for performance. For more details, read
                             https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
2024-07-16 10:18:53,576 - lightning.pytorch.accelerators.cuda - INFO - LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
[07/16/24 10:18:53] INFO     LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]                                                                                                                                           cuda.py:58
┏━━━┳━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━┓
┃   ┃ Name                  ┃ Type                     ┃ Params ┃
┡━━━╇━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━┩
│ 0 │ model                 │ DraemModel               │ 97.4 M │
│ 1 │ loss                  │ DraemLoss                │      0 │
│ 2 │ _transform            │ Compose                  │      0 │
│ 3 │ normalization_metrics │ MinMax                   │      0 │
│ 4 │ image_threshold       │ F1AdaptiveThreshold      │      0 │
│ 5 │ pixel_threshold       │ F1AdaptiveThreshold      │      0 │
│ 6 │ image_metrics         │ AnomalibMetricCollection │      0 │
│ 7 │ pixel_metrics         │ AnomalibMetricCollection │      0 │
└───┴───────────────────────┴──────────────────────────┴────────┘
Trainable params: 97.4 M
Non-trainable params: 0
Total params: 97.4 M
Total estimated model params size (MB): 389
/home/jinyao/anaconda3/envs/IADBE/lib/python3.10/site-packages/lightning/pytorch/core/module.py:494: You called `self.log('train_loss', ..., logger=True)` but have no logger configured. You can enable one by doing
`Trainer(logger=ALogger(...))`
Epoch 20/699 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18/18 0:00:17 • 0:00:00 1.06it/s train_loss_step: 0.772 pixel_AUROC: 0.334 pixel_PRO: 1.000 train_loss_epoch: 0.8052024-07-16 10:28:07,898 - anomalib.callbacks.timer - INFO - Training took 554.31 seconds
[07/16/24 10:28:07] INFO     Training took 554.31 seconds                                                                                                                                                       timer.py:59
Epoch 20/699 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18/18 0:00:17 • 0:00:00 1.06it/s train_loss_step: 0.772 pixel_AUROC: 0.531 pixel_PRO: 0.092 train_loss_epoch: 0.782
2024-07-16 10:28:07,904 - anomalib.data.image.kolektor - INFO - Found the dataset.
                    INFO     Found the dataset.                                                                                                                                                             kolektor.py:340
2024-07-16 10:28:08,081 - lightning.pytorch.accelerators.cuda - INFO - LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
[07/16/24 10:28:08] INFO     LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]                                                                                                                                           cuda.py:58
Testing ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8/8 0:00:21 • 0:00:00 0.33it/s  2024-07-16 10:28:34,175 - anomalib.callbacks.timer - INFO - Testing took 25.46011185646057 seconds
Throughput (batch_size=16) : 4.791809269645537 FPS
[07/16/24 10:28:34] INFO     Testing took 25.46011185646057 seconds                                                                                                                                            timer.py:109
                             Throughput (batch_size=16) : 4.791809269645537 FPS
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │    0.5192307233810425     │
│         image_PRO         │            1.0            │
│        pixel_AUROC        │    0.5313490629196167     │
│         pixel_PRO         │    0.0923960730433464     │
└───────────────────────────┴───────────────────────────┘
Testing ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8/8 0:00:21 • 0:00:00 0.33it/s

